### PR TITLE
Set minimum PyTorch version to 1.6

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title.

**TODO**: Make sure the "Current build statuses" table in the README is correct

Screenshot 2021-07-05 at 14 11 03

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/ci_dockers.yml [hunk 1]
```diff
@@ -68,33 +68,26 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          # todo: see notes in Dockerfile
           - python_version: 3.6
-            pytorch_version: 1.4
+            pytorch_version: 1.6
           - python_version: 3.7
             pytorch_version: 1.7
```
Comment (Borda) [inline .github/workflows/ci_dockers.yml:74]:
```suggestion
```

File: .github/workflows/ci_dockers.yml [hunk 2]
```diff
@@ -68,33 +68,26 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          # todo: see notes in Dockerfile
           - python_version: 3.6
```
Comment (Borda) [inline .github/workflows/ci_dockers.yml:71]:
```suggestion
          - python_version: 3.7
```

File: .github/workflows/ci_dockers.yml [hunk 3]
```diff
@@ -106,7 +99,7 @@ jobs:
       matrix:
         include:
           - python_version: 3.6
-            pytorch_version: 1.4
+            pytorch_version: 1.6
           - python_version: 3.7
             pytorch_version: 1.7
```
Comment (Borda) [inline .github/workflows/ci_dockers.yml:104]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Comment by tchaton** [comment]
Hey @ananthsub, 

Would you mind to validate there is no specific reason for us to keep supporting 1.4 and 1.5 ?

Best,
T.C

**Comment by ananthsub** [comment]
> Hey @ananthsub,
> 
> Would you mind to validate there is no specific reason for Lightning to keep supporting 1.4 and 1.5 ?
> 
> Best,
> T.C

As discussed today, this is primarily motivated by how long the backward-compatibility window ought to be. Users can always pin themselves to older versions of Lightning if they absolutely need older PyTorch support. Given how stable the PyTorch API has been, there's less risk of breaking changes across these versions that Lightning needs to cover for. However there are a ton of features users will get with both PyTorch and Lightning by updating to recent versions. Plus CI time/$ savings for contributors, but this is a minor concern compared to user adoption.

@carmocca @Borda if you want to be more conservative, we could start with not supporting PyTorch v1.4, see how the community receives this, and then determine if this is a pattern we can continue for the next releases (e.g. always support the latest N releases of PyTorch, or PyTorch versions within last T time).

**Comment by Borda** [comment]
I may say that with adding support for future PT 1.10 #8133 we can do a bit bigger step and drop also PT 1.5 for the upcoming PL 1.4

**Comment by tchaton** [comment]
> > we could start with not supporting PyTorch v1.4, see how the community receives this
> 
> I may say that with adding support for future PT 1.10 #8133 we can do a bit bigger step and drop also PT 1.5 for the upcoming PL 1.4 (so we would have still 5 minor PT support)

Yes, I think it is a good approach.

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Pull request merged**